### PR TITLE
Use KDE SDK, Runtime and KF6

### PR DESCRIPTION
**Pull request by sungsphinx**
> [!IMPORTANT]  
> This needs testing, please do not mark as ready ***yet***.

Uses changes from:

https://github.com/flathub/org.libreoffice.LibreOffice/pull/253 (Switch to KDE Runtime/SDK and Building with KF5/QT5 VCL's Backends by @EliasOfWaffle, based on @CarlSchwan's PR)

### CODE REVIEW

File: org.libreoffice.LibreOffice.json [hunk 1]
```diff
@@ -814,8 +826,16 @@
...
+                "append-path": "/usr/lib/sdk/llvm16/bin",
+                "prepend-ld-library-path": "/usr/lib/sdk/llvm16/lib",
+                "env": {
+                    "QT6DIR": "/usr"
+                }
+             },
             "build-commands": [
-                "./autogen.sh --prefix=/run/build/libreoffice/inst --with-distro=LibreOfficeFlatpak --with-gdrive-client-id=280867422816-9jc83t6phkfb2q8p94dtk8kr5fa8af3r.apps.googleusercontent.com --with-gdrive-client-secret=Hnzikj7HqdqsUYLru4jmFo1p",
+                "source /usr/lib/sdk/llvm17/enable.sh",
+                "./autogen.sh CC=clang CXX=clang++ --prefix=/run/build/libreoffice/inst --with-distro=LibreOfficeFlatpak --enable-kf6 --enable-qt6 --with-gdrive-client-id=280867422816-9jc83t6phkfb2q8p94dtk8kr5fa8af3r.apps.googleusercontent.com --with-gdrive-client-secret=Hnzikj7HqdqsUYLru4jmFo1p",
```
Comment (TheFlagCourier) [inline org.libreoffice.LibreOffice.json:838]: Your build is failing due to clang not finding standard headers. On that note, is there a particular reason to use clang?

> --enable-ld=\<linker\>  |  Use the specified linker. Both 'gold' and 'lld' linkers generally use less memory and link faster. By default tries to use the best linker possible, use --disable-ld to use the default linker. If \<linker\> contains any ':', the part before the first ':' is used as the value of -fuse-ld, while the part after the first ':' is used as the value of --ld-path (which is needed for Clang 12).

**Review by TheFlagCourier — COMMENTED** [review]
Took a cursory glance at this, since I would like better compatibility on the Steam Deck and Fedora 40 Plasma. Questions for two things in the review.

**Comment by EliasTheGrandMasterOfMistakes** [comment]
Is not better to use qt6 and KF as dependency for the default runtime instead of change this?

**Comment by EliasTheGrandMasterOfMistakes** [comment]
This is the same build error from older kde runtime

**Pull request closed without merging**